src/cmd/sign.js

```diff
@@ -155,8 +155,13 @@ export default function sign(
       let result;
       try {
         if (useSubmissionApi) {
-          // $FlowIgnore: we verify 'channel' is set above
-          result = await submitAddon({...signSubmitArgs, amoBaseUrl, channel});
+          result = await submitAddon({
+            ...signSubmitArgs,
+            amoBaseUrl,
+            // $FlowIgnore: we verify 'channel' is set above
+            channel,
+            saveIdFunc: (genId: string) => saveIdToSourceDir(sourceDir, genId),
```

Comment:
Is this _only_ needed because `saveIdToSourceDir()` is declared in this file?

Instead of passing a function as a parameter, I would rather call `saveIdToSourceDir()` in the `submitAddon()` module directly personally.

Reply:
It also means we don't need to pass down the source dir. Having the function as the parameter seemed to be a cleaner change than passing down the source dir just for that, and moving the `saveIdToSourceDir` function elsewhere. But I'm also not clear where you draw the line between functionality that's supposed to be in `sign.js` v.s. `submit-addon.js`